src/main/scala/rifs/business/controllers/ApplicationController.scala

```diff
@@ -46,8 +47,77 @@ class ApplicationController @Inject()(val cached: Cached, applications: Applicat
     applications.saveSection(id, sectionNumber, request.body).map(_ => NoContent)
   }
 
-  def completeSection(id: ApplicationId, sectionNumber: Int) = Action.async(parse.json[JsObject]) { implicit request =>
-    applications.saveSection(id, sectionNumber, request.body, Some(LocalDateTime.now())).map(_ => NoContent)
+  def deleteSectionItem(id: ApplicationId, sectionNumber: Int, itemNumber: Int) = Action.async { implicit request =>
+    def hasItemNumber(o: JsObject, num: Int) = o \ "itemNumber" match {
+      case JsDefined(JsNumber(n)) if n == num => true
+      case _ => false
+    }
+
+    applications.fetchAppWithSection(id, sectionNumber).flatMap {
+      case Some((app, os)) =>
+        val doc = os.map(_.answers).getOrElse(JsObject(Seq()))
+        val items = doc \ "items" match {
+          case JsDefined(JsArray(is)) => is.collect { case o: JsObject => o }
+          case _ => Seq()
+        }
+        val remainingItems = items.filterNot(o => hasItemNumber(o, itemNumber))
+        val updated = doc + ("items" -> JsArray(remainingItems))
+        applications.saveSection(id, sectionNumber, updated).map(_ => NoContent)
+
+      case None => Future.successful(NotFound)
+    }
+  }
+
+  def putSectionItem(id: ApplicationId, sectionNumber: Int, itemNumber: Int) = Action.async(parse.json[JsObject]) { implicit request =>
```

Comment:
There is a lot similarity with "deleteSectionItem" code here. 
Should we have something like 
def forSectionItem(appId, sectionNum)(func: Section=>Future[X]) ?
to avoid repeating of the code?